packages/core/src/components/document-component/document-component.tsx

```diff
@@ -183,7 +206,9 @@ export class DocumentComponent implements ComponentInterface {
   }
 
   private handleHeadingClick = (e: MouseEvent) => {
-    if (this.type !== "LID") {
+    if (this.mode === "table-of-contents") {
+      this.dsoTableOfContentsClick.emit({ originalEvent: e, isModifiedEvent: isModifiedEvent(e) });
+    } else if (this.type !== "LID") {
```

Comment:
Heb je gekeken naar de mogelijkheden om de JSX binding `<a onClick={}>` van de betreffende Anchor te gebruiken?

Als ik het goed begrijp: Anchor clicks in Noten worden nu ge-emit als TableOfContentsClick?

Reply:
Mijn voornamelijkste bezwaar is dat er geen `onClick` op de `<a>` zit: https://github.com/dso-toolkit/dso-toolkit/pull/2888/files#diff-baa1a6e625b438ba6cf9f30c8e35098c0e9408af8ea48c5eeab8d73c6aa5ef45R39

We doen een luie truuk met de `{...props}` spread operator waar `props: DocumentComponentHeadingProps & JSXBase.HTMLAttributes<HTMLHeadingElement>`, maar nu het functionele component Heading geen wrapper meer voor `<h1>-<h6>` is, moeten we dit wat robuster opzetten.

Reply:
Is het nu dan voldoende dat ik als fallback het headingElement retourneer?

Reply:
Nee, de fallback heeft hier niets mee te maken.

Maar je hebt wel code gewijzigd die weer voor nieuwe feedback zorgt 😄 

Er zijn nu 3 problemen:

1. Er worden properties op attributen gezet die er niet horen: 
![image](https://github.com/user-attachments/assets/a8a9c0e6-b071-464a-95b3-12f6fda0db21)
2. De anchor activeren met het toetsenbord (Focus en `ENTER`) zorgt ervoor dat er geen `dsoTableOfContentsClickEvent` wordt afgevuurd, maar de anchor wordt gevolgd. In Storybook uit zich dat als volgt:
![image](https://github.com/user-attachments/assets/c43cc2d9-2412-4243-baf5-f439b75195a9)
3. Het element in `dsoTableOfContentsClick.detail.currentTarget` zou naar de anchor moeten verwijzen (want dat is het interactieve element), maar verwijst naar het heading element:
<img width="410" alt="image" src="https://github.com/user-attachments/assets/66346b26-e32d-4a64-b916-7589bad67fa0">

De oorzaken zijn:

1. `href` en `mode` zijn onderdeel van de `props` van het functionele component `Header` en alle props behalve `heading` worden gespread: `<h123456 {...props}>`.
2. (en 3) De event listener zit op het heading (`<h123456>`) element en niet op de anchor (`<a>`)

Het functionele component `<Header>` moet uitgebreid worden op de situatie dat het ook met een `<a>` kan werken. Wat ik net "de luie manier" noemde was zoals het voorheen was opgezet: Er kwam _altijd_ en heading (`<h123456>`) uit `<Header />` waardoor het logisch was om de props dezelfde vorm te laten zijn. Maar nu er ook een `<a>` uit kan komen, moeten we iets specifieker worden.

Van je nieuwe opzet vind ik het leuk dat je `href` en `mode` hebt toegevoegd, hoewel ik voor `href: string | undefined` was gegaan ipv `href?: string`: De typings zijn hetzelfde maar de betekenis is anders. `href` is niet optioneel maar verplicht voordat `<Heading>` op de juiste manier kan bepalen of de headings (`<h123456>`) gewrapped wordt in een `<a>`.

Ongeteste suggestie:

```ts
export const Heading: FunctionalComponent<
  DocumentComponentHeadingProps & JSXBase.HTMLAttributes<HTMLHeadingElement>
> = ({ heading, mode, href, onClick, ...props }, children) => {
  let headingElement: VNode;
  switch (heading) {
    default:
    case "h2":
      headingElement = <h2 {...props} onClick={onClick}>{children}</h2>;
      break;
    case "h3":
      headingElement = <h3 {...props} onClick={onClick}>{children}</h3>;
      break;
    case "h4":
      headingElement = <h4 {...props} onClick={onClick}>{children}</h4>;
      break;
    case "h5":
      headingElement = <h5 {...props} onClick={onClick}>{children}</h5>;
      break;
    case "h6":
      headingElement = <h6 {...props} onClick={onClick}>{children}</h6>;
  }

  if (mode === "table-of-contents" && href) {
    return (
      <a href={href} onClick={onClick} class="heading-anchor">
        {headingElement}
      </a>
    );
  }

  return headingElement;
};
```

In mijn eerste comment zeg ik nog:

> Als ik het goed begrijp: Anchor clicks in Noten worden nu ge-emit als TableOfContentsClick?

Maar dat is volgens mij onzin: Ik ben in de war met `private handleOzonContentAnchorClick`.

Kan je hier iets mee?

Reply:
Ik denk het wel. Ik heb het gedaan.

Reply:
Ok, dan denk ik dat we er zijn